mlibrary/pride
# Updating `eslint` and rebuilding the setup.

# Overview
`dependabot` has made its first update, which exposed `eslint-config-standard` not being up to date in over a year. It is not compatible with the latest version of [`eslint`](https://www.npmjs.com/package/eslint). Turns out that initializing `eslint` has changed. This pull request uninstalls all `eslint`-related dependencies and follows the new setup, which added these dependencies:
- `@eslint/js`
- `eslint-plugin-mocha`
- `globals`

These dependencies have been removed:
- `eslint-config-standard`
- `eslint-plugin-import`
- `eslint-plugin-n`
- `eslint-plugin-promise`

## Testing
1. Build the repository (`npm run build`).
2. Run the tests to make sure they pass (`npm run test`).
   * Break the new/updated unit tests to make sure they're working properly.
3. Navigate to your local [Search repository](https://github.com/mlibrary/search) and apply the newly generated files:
   1. Open `./package.json` and add `#branch-name-of-pull-request` at the end of the `pride` dependency URL:
      ```bash
      "pride": "git+https://github.com/mlibrary/pride.git#branch-name-of-pull-request"
      ``` 
   2. Do a clean install of Search:
      ```bash
      rm -rf node_modules && rm package-lock.json && npm install
      ``` 
   3. Start Search (`npm start`) and look around [the site](http://localhost:3000/everything). Check to see if there are any console errors, and everything still works as expected.
- Make sure the PR is consistent in these browsers:
